sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JSONOptions.scala

```diff
@@ -59,6 +62,8 @@ object JSONOptions {
     allowNumericLeadingZeros =
       parameters.get("allowNumericLeadingZeros").map(_.toBoolean).getOrElse(false),
     allowNonNumericNumbers =
-      parameters.get("allowNonNumericNumbers").map(_.toBoolean).getOrElse(true)
+      parameters.get("allowNonNumericNumbers").map(_.toBoolean).getOrElse(true),
```

Comment:
I'm suggesting changing the default value to true, and also update all the documentations (as you pointed out) to true.